test/integration/targets/azure_rm_virtualmachine/basic.yaml

```diff
@@ -0,0 +1,293 @@
+  azure_rm_networkinterface:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_nics[0] }}"
+    virtual_network: "{{ hostvars['localhost'].vnet }}"
+    subnet: "{{ hostvars['localhost'].subnet }}"
+    security_group: "{{ hostvars['localhost'].nsg }}"
+    public_ip_name: "{{ hostvars['localhost'].pip }}"
+
+- name: Create second NIC
+  azure_rm_networkinterface:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_nics[1] }}"
+    virtual_network: "{{ hostvars['localhost'].vnet }}"
+    subnet: "{{ hostvars['localhost'].subnet }}"
+    security_group: "{{ hostvars['localhost'].nsg }}"
+
+# Test VM basics; Legacy storage, nic creation and removal of resources post VM delete
+- name: Create virtual machine with a single NIC and legacy storage
+  azure_rm_virtualmachine:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_names.basic }}"
+    short_hostname: "{{ hostvars['localhost'].vm_names.basic }}"
+    admin_username: "{{ hostvars['localhost'].admin_user }}"
+    admin_password: "{{ hostvars['localhost'].password }}"
+    availability_set: "{{ hostvars['localhost'].availability_set_blob }}"
+    network_interface_names: "{{ hostvars['localhost'].vm_nics[0] }}"
+    storage_account: "{{ hostvars['localhost'].storage_account }}"
+    storage_container: "{{ hostvars['localhost'].vm_names.basic }}"
+    storage_blob: "{{ hostvars['localhost'].vm_names.basic }}.vhd"
+    os_type: "{{ hostvars['localhost'].os_type }}"
+    image: "{{ hostvars['localhost'].image }}"
+    vm_size: "{{ hostvars['localhost'].vm_size }}"
+    custom_data: |
+      #!/bin/sh
+      echo "{{ hostvars['localhost'].target_custom_data }}" > "{{ hostvars['localhost'].target_custom_data_file }}"
+  register: output
+
+- assert:
+    that:
+      - azure_vm.properties.availabilitySet.id
+
+- name: Should be idempotent with a single NIC and legacy storage
+  azure_rm_virtualmachine:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_names.basic }}"
+    short_hostname: "{{ hostvars['localhost'].vm_names.basic }}"
+    admin_username: "{{ hostvars['localhost'].admin_user }}"
+    admin_password: "{{ hostvars['localhost'].password }}"
+    availability_set: "{{ hostvars['localhost'].availability_set_blob }}"
+    network_interface_names: "{{ hostvars['localhost'].vm_nics[0] }}"
+    storage_account: "{{ hostvars['localhost'].storage_account }}"
+    storage_container: "{{ hostvars['localhost'].vm_names.basic }}"
+    storage_blob: "{{ hostvars['localhost'].vm_names.basic }}.vhd"
+    os_type: "{{ hostvars['localhost'].os_type }}"
+    image: "{{ hostvars['localhost'].image }}"
+    vm_size: "{{ hostvars['localhost'].vm_size }}"
+  register: output
+
+- assert:
+    that: not output.changed
+    
+- debug:
+    var: output
+
+- add_host:
```

Comment:
This is causing integration test failures, as add_host cannot be used with the free strategy:

https://app.shippable.com/github/ansible/ansible/runs/81844/73/console

Which means we won't be able to run this part of the test in parallel.

Reply:
I am sorry I do not know enough to get this right. Can you complete this please. I added debug to look at the output as I had trouble resolving the IP address in the past with this test.